tests/rptest/tests/topic_recovery_test.py

```diff
@@ -164,10 +186,10 @@ def get_ntp_sizes(fdata_per_host, hosts_can_vary=True):
         rest_ntp_size = restored_ntps[ntp]
         assert rest_ntp_size <= orig_ntp_size, f"NTP {ntp} the restored partition is larger {rest_ntp_size} than the original one {orig_ntp_size}."
         delta = orig_ntp_size - rest_ntp_size
+        tolerance = int(1.5 * default_log_segment_size)
+        tolerance = max(tolerance, baseline_last_segment_sizes.get(ntp, 0))
```

Comment:
Now that we have the exact last segment size, do we still need the 1.5 tolerance?

Maybe there is still some tolerance needed (if the restored system has e.g. created an extra empty segment), but probably not the full 1.5?

Reply:
We probably don't need the default any more because we filter out the empty segments when getting ntp sizes.  I had added it as a fallback but I think we should be okay removing it completely.

Reply:
yep, that should be OK

Reply:
it looks like this test is consistently failing on CI when we rely on the last segment size. Sometimes the 1.5x default segment size is greater than last segment. Will look through the logs.